### PR TITLE
Neodigit.net DNS API

### PR DESCRIPTION
**Pull request by aalmenar**
Hi,

This add neodigit.net DNS API for use inside acme.sh.

### CODE REVIEW

File: dnsapi/dns_neodigit.sh [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+NEODIGIT_API_URL="https://api.neodigit.net/v1"
+#
+########  Public functions #####################
+
+# Usage: dns_myapi_add _acme-challenge.www.domain.com "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_neodigit_add() {
+  fulldomain=$1
+  txtvalue=$2
+
+  NEODIGIT_API_TOKEN="${NEODIGIT_API_TOKEN:-$(_readaccountconf_mutable NEODIGIT_API_TOKEN)}"
```
Comment (Neilpang) [inline dnsapi/dns_neodigit.sh]: just use:
```
NEODIGIT_API_TOKEN="$(_readaccountconf_mutable NEODIGIT_API_TOKEN)"
```

File: dnsapi/dns_neodigit.sh [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+  _debug "Getting txt records"
+  _neo_rest GET "dns/zones/${_domain_id}/records?type=TXT&name=$fulldomain&content=$txtvalue"
+
+  if [ "$_code" != "200" ]; then
+    _err "error retrieving data!"
+    return 1
+  fi
+
+  record_id=$(printf "%s\n" "$response" | _egrep_o "\"id\":\s*[0-9]+" | _head_n 1 | cut -d: -f2 | cut -d, -f1)
```
Comment (Neilpang) [inline dnsapi/dns_neodigit.sh]: you can use "echo" instead of "printf "  here

File: dnsapi/dns_neodigit.sh [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    fi
+
+    if ! _neo_rest GET "dns/zones?name=$h"; then
+      return 1
+    fi
+
+    _debug p "$p"
+
+    if _contains "$response" "\"name\":\"$h\"" >/dev/null; then
+      _domain_id=$(printf "%s\n" "$response" | _egrep_o "\"id\":\s*[0-9]+" | _head_n 1 | cut -d: -f2 | cut -d, -f1)
```
Comment (Neilpang) [inline dnsapi/dns_neodigit.sh]: you can use "echo" instead of "printf "

File: dnsapi/dns_neodigit.sh (hunk 2 above)
Comment (aalmenar) [reply dnsapi/dns_neodigit.sh]: Its based on dns_cf which uses printf also...

Comment (aalmenar) [reply dnsapi/dns_neodigit.sh]: Updated pull request.

File: dnsapi/dns_neodigit.sh (hunk 3 above)
Comment (aalmenar) [reply dnsapi/dns_neodigit.sh]: Updated pull request.

File: dnsapi/dns_neodigit.sh [hunk 4]
```diff
@@ -17,7 +17,7 @@ dns_neodigit_add() {
   fulldomain=$1
   txtvalue=$2
 
-  NEODIGIT_API_TOKEN="${NEODIGIT_API_TOKEN:-$(_readaccountconf_mutable NEODIGIT_API_TOKEN)}"
+  NEODIGIT_API_TOKEN="$(_readaccountconf_mutable NEODIGIT_API_TOKEN)}"
```
Comment (Neilpang) [inline dnsapi/dns_neodigit.sh]: sorry, it's my fault. please revert this change to your first version.

Comment (aalmenar) [reply dnsapi/dns_neodigit.sh]: Done.
